static/js/list_render.js

```diff
@@ -320,3 +320,43 @@ var list_render = (function () {
 if (typeof module !== 'undefined') {
     module.exports = list_render;
 }
+
+$(function () {
+    /*
+    one would specify sort parameters like this:
+        - name => sort alphabetic.
+        - age  => sort numeric.
+
+    you MUST specify the `data-list-render` in the `.progressive-table-wrapper`
+    otherwise it will not know what `list_render` instance to look up.
+
+    <div class="progressive-table-wrapper" data-list-render="some-list">
+        <table>
+            <tr>
+                <td data-sort="alphabetic" data-sort-prop="name">
+                <td data-sort="numeric" data-sort-prop="age">
+            </tr>
+        </table>
+    </div>
+    */
+    $("body").on("click", "[data-sort]", function () {
+        var $this = $(this);
+        var sort_type = $this.data("sort");
+        var prop_name = $this.data("sort-prop");
+        var list_name = $this.closest(".progressive-table-wrapper").data("list-render");
+
+        var list = list_render.get(list_name);
+
+        if (!list) {
+            blueslip.warn("Error. This `.progressive-table-wrapper` has no `data-list-render` attribute.");
+            return;
+        }
+
+        if (prop_name) {
+            list.sort(sort_type, prop_name);
+            return;
+        }
+
+        list.sort(sort_type);
```

Comment:
I'm not sure why you need 2 cases here; can't you just call `list.sort(sort_type, prop_name)` with `prop_name` being potentially undefined?

Reply:
This really has to do with #6759, this PR is only supposed to be the last commit on top of #6759 but it's on top for testing. I fixed both of your concerns though.